modules/contracts/src.ts/services/ethService.ts

```diff
@@ -97,6 +101,44 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     if (!channelState.latestUpdate.aliceSignature || !channelState.latestUpdate.bobSignature) {
       return Result.fail(new ChainError(ChainError.reasons.MissingSigs));
     }
+
+    const code = await this.getCode(channelState.channelAddress, channelState.networkContext.chainId);
+    if (code.isError) {
+      return Result.fail(code.getError()!);
+    }
+    if (code.getValue() === "0x") {
```

Comment:
what is it about this condition that prompts us to deploy the channel? why do we deploy a channel just to defund it due to a dispute?

Reply:
You need the channel deployed to call any of the methods on the contracts. If it's not deployed it's just an address.